### ISSUE TITLE
Pre selecting QNH whilst in Std does not respect the baro unit selection knob

### ISSUE THREAD

**Issue by ShiforGitHub222** [flybywiresim/aircraft#9291]
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-10-31T15:54:41+00:00",
    "ref": "refs/tags/v0.12.0",
    "sha": "2fe89113291f2478347dde0dc3228b55dc924f74",
    "actor": "FoxtrotSierra6829",
    "event_name": "manual",
    "pretty_release_name": "stable/v0.12.0",
    "version": "a380x-v0.12.0-rel.2fe8911"
}
```

### Describe the bug

Pre selecting the QNH whilst in Std does not respect the baro unit selection knob.

If you switch back out of Std, select the desired unit, go back into Std and then try to pre select a QNH/altimeter, it'll use the previously selected unit. 

### Expected behavior

When pre selecting the QNH, I expect it to be in the format currently selected on the baro knob (inHg or hPa).

### Steps to reproduce

1. Switch to Std pressure
2. Change QNH units (to inHg or HpA)
3. Scroll baro knob to pre select QNH
4. Observe it not respecting the unit selected

### References (optional)

https://github.com/user-attachments/assets/df8d8e1f-f7fe-4370-9213-8571639d5122

### Discord Username (optional)

Shifor

**Comment by tracernz** [comment]
Duplicate of https://github.com/flybywiresim/aircraft/issues/9161 and already fixed.

**Comment by Benjozork** [comment]
This fix will be available in `v0.12.1`.
